packages/react-scripts/config/config.js

```diff
@@ -0,0 +1,110 @@
+// @remove-on-eject-begin
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const paths = require('./paths');
+
+const appDirectory = fs.realpathSync(process.cwd());
+
+/**
+ * Get the baseUrl of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getBaseUrl(options = {}) {
+  const baseUrl = options.baseUrl;
+
+  if (!baseUrl) {
+    return null;
+  }
+
+  if (baseUrl !== 'src' && baseUrl !== './src') {
```

Comment:
This needs to support `node_modules` as well, which is the "default value".

You cannot set `paths` without `baseUrl`, so users who want to configure only aliases will want to keep the default behavior of `node_modules`.

Reply:
https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping

![image](https://user-images.githubusercontent.com/616428/50743949-9f5a6900-11eb-11e9-8a28-dffac65565f8.png)

Reply:
For reference, this is how I have this configures on a fork of react-scripts (for a client project) - it works well, and let's me import (as an example) `../../utils/xyz` as `utils/xyz`.
```
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "*": [
        "src/*"
      ]
    }
  },
```

Reply:
Hmm, the current implementation would not support syntax like `src/*`. Is there any differences between that and just `src`?

Reply:
If you remove the `/*` TypeScript will fail to resolve. However, as we're currently supporting `src`-only (as I understand it) the config you've provided is good.

If we supported other directories, we could use an approach like this:
```
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "some-module": ["not-src/some-module"]
      "*": ["src/*"],
      "@*": ["src/*"]
    }
  },
```

Reply:
Hmm, but we'll want consistent behavior between editor & CRA so we should probably support this `/*`.